course-info/src/fetch-exam.ts

```diff
@@ -16,18 +16,11 @@ async function fetchExamText(url: string): Promise<string> {
 }
 
 function parsePrelimLine(line: string): ExamInfo {
-  const segments = line.split(' ');
+  const segments = line.split(/\s+/);
   const subject = segments[0];
   const courseNumber = segments[1];
-  let datetimeString = '';
-  for (let i = 2; i < segments.length; i += 1) {
-    const s = segments[i];
-    if (s !== '') {
-      datetimeString = s;
-      break;
-    }
-  }
-  const date = new Date(datetimeString);
+  const dateTimeString = `${segments[3]} ${segments[4]} ${currentYear}`;
+  const date = new Date(dateTimeString);
   date.setFullYear(currentYear);
   date.setHours(19, 30); // 7:30 prelim time
```

Comment:
the screenshot shows some exams at 4:00PM, so I guess we should somehow change this?

Reply:
Done.